tests/check_check_tags.c

```diff
@@ -0,0 +1,621 @@
+    ck_assert(!test_executed[YELLOW]);
+    ck_assert(!test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(inc_yellow)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, "Yellow", NULL, CK_VERBOSE);
+
+    ck_assert(!test_executed[RED]);
+    ck_assert(!test_executed[BLUE]);
+    ck_assert(!test_executed[PURPLE]);
+    ck_assert(test_executed[YELLOW]);
+    ck_assert(!test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+
+START_TEST(inc_red)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, "Red", NULL, CK_VERBOSE);
+
+    ck_assert(test_executed[RED]);
+    ck_assert(!test_executed[BLUE]);
+    ck_assert(test_executed[PURPLE]);
+    ck_assert(!test_executed[YELLOW]);
+    ck_assert(!test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(inc_red_blue)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, "Red Blue", NULL, CK_VERBOSE);
+
+    ck_assert(test_executed[RED]);
+    ck_assert(test_executed[BLUE]);
+    ck_assert(test_executed[PURPLE]);
+    ck_assert(!test_executed[YELLOW]);
+    ck_assert(!test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(inc_red_blue_yellow)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, "Red Blue Yellow", NULL, CK_VERBOSE);
+
+    ck_assert(test_executed[RED]);
+    ck_assert(test_executed[BLUE]);
+    ck_assert(test_executed[PURPLE]);
+    ck_assert(test_executed[YELLOW]);
+    ck_assert(!test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(exc_yellow)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, NULL, "Yellow", CK_VERBOSE);
+
+    ck_assert(test_executed[RED]);
+    ck_assert(test_executed[BLUE]);
+    ck_assert(test_executed[PURPLE]);
+    ck_assert(!test_executed[YELLOW]);
+    ck_assert(test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(exc_red)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, NULL, "Red", CK_VERBOSE);
+
+    ck_assert(!test_executed[RED]);
+    ck_assert(test_executed[BLUE]);
+    ck_assert(!test_executed[PURPLE]);
+    ck_assert(test_executed[YELLOW]);
+    ck_assert(test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(exc_red_blue)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, NULL, "Red Blue", CK_VERBOSE);
+
+    ck_assert(!test_executed[RED]);
+    ck_assert(!test_executed[BLUE]);
+    ck_assert(!test_executed[PURPLE]);
+    ck_assert(test_executed[YELLOW]);
+    ck_assert(test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(exc_red_blue_yellow)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, NULL, "Red Blue Yellow", CK_VERBOSE);
+
+    ck_assert(!test_executed[RED]);
+    ck_assert(!test_executed[BLUE]);
+    ck_assert(!test_executed[PURPLE]);
+    ck_assert(!test_executed[YELLOW]);
+    ck_assert(test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+
+START_TEST(inc_red_exc_red)
```

Comment:
So, the understanding of the include and exclude parameters is that the exclude will always prevail? Just make this clear in the documentation.